scripts/generate/test_generate.py

```diff
@@ -147,6 +141,8 @@ def test_generate(
         # sequences would require https://github.com/pytorch/torchtitan/pull/686
         apply_tp_minus_sp(model, world_mesh["tp"])
 
+    utils.set_determinism(world_mesh, device, seed, deterministic=(seed is not None))
```

Comment:
"seed is not None" is equiavlent to the old logic, but i still think its better to separate the concepts. 

it seems mandatory to use a seed for the token sampler so that each rank uses the same token.   However, it should be optional to force kernels to run in the slower 'deterministic' mode.  So it should be possible to run with a seed but not assume deterministic=True